python/cudf/cudf/core/dataframe.py

```diff
@@ -831,8 +831,12 @@ def _init_from_dict_like(
                 self._data.multiindex = self._data.multiindex and isinstance(
                     col_name, tuple
                 )
-                self.insert(
-                    i, col_name, data[col_name], nan_as_null=nan_as_null
+                self._insert(
+                    i,
+                    col_name,
+                    data[col_name],
+                    nan_as_null=nan_as_null,
+                    ignore_index=True,
```

Comment:
```suggestion
                    nan_as_null=nan_as_null,
```

Reply:
Somehow I can't add suggestion to batch or commit your suggestion in this review 😕 so pushed separate commit.